### ISSUE TITLE
[Feature Request]: Add mask management to BatchNormalization layer

### ISSUE THREAD

**Issue by Utanapishtim31** [SciSharp/TensorFlow.NET#1198]
### Background and Feature Description

BatchNormalization output is sensitive to masking. Masking should be handled in the same way as RNN layers.

The mask tensor should be detected as an input of the BatchNormalization layer by the dependency detection mechanism to avoid the current exceptions raised when a Mask is declared for RNN layers. The same fix should be done for the RNN layers.

### API Definition and Usage

```cs
public class BatchNormalizationOptionalArgs : IOptionalArgs
{
    public string Identifier => "BatchNormalization";
    public Tensor Mask { get; set; }
}
```

### Risks

Minimal
